components/trello/cards-due.js

```diff
@@ -0,0 +1,67 @@
+const trello = require("https://github.com/PipedreamHQ/pipedream/components/trello/trello.app.js");
+
+module.exports = {
+  name: "Cards Due",
+  description:
+    "Emits an event at a specified time before a card is due.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    trello,
+    boardId: { propDefinition: [trello, "boardId"] },
+    timeBefore: {
+      type: "integer",
+      label: "Time Before",
+      description: "How far before the due time the event should trigger.",
+      default: 5,
+    },
+    timeBeforeUnit: {
+      type: "string",
+      label: "Time Before (Unit)",
+      description: "Unit of time for Time Before.",
+      async options() {
+        return [
+          { label: 'Minutes', value: 60000 },
+          { label: 'Hours', value: 3600000 },
+          { label: 'Days', value: 86400000 },
+          { label: 'Weeks', value: 604800000 },
+        ]
+      },
+    },
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+
+  async run(event) {
+    let cards = [];
+    let results = [];
+    let due, notifyAt, n;
+    const modelId = this.boardId ? this.boardId : "me";
```

Comment:
You use a different pattern in `card-updates.js` to get the model id using `await this.trello.getMember("me");` in the absence of a board Id - do you need that logic here, too?

Reply:
In this file, modelId is being used to poll for the cards on a board, but you can use "me" as the id to get all of the user's boards. I probably should have called this boardId.
In card-updates.js, the modelId is being used to create a webhook which doesn't allow using "me" as the id. So if there's no boardId, it calls getMember("me") to retrieve the user's member id.